library/core/testing/src/main/java/org/quiltmc/qsl/testing/mixin/command/TestCommandMixin.java

```diff
@@ -0,0 +1,118 @@
+			),
+			index = 1
+	)
+	private static ArgumentType<String> quiltGameTest$replaceExportImportTestNameArgumentType(ArgumentType<String> name) {
+		return new TestNameArgumentType();
+	}
+
+	@ModifyArg(
+			method = "register",
+			slice = @Slice(
+					from = @At(value = "CONSTANT", args = "stringValue=create"),
+					to = @At(value = "CONSTANT", args = "stringValue=width")
+			),
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/server/command/CommandManager;argument(Ljava/lang/String;Lcom/mojang/brigadier/arguments/ArgumentType;)Lcom/mojang/brigadier/builder/RequiredArgumentBuilder;"
+			),
+			index = 1
+	)
+	private static ArgumentType<String> quiltGameTest$replaceCreateTestNameArgumentType(ArgumentType<String> name) {
+		return new TestNameArgumentType();
+	}
+
+	@Redirect(
+			method = "run(Lnet/minecraft/server/world/ServerWorld;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/test/TestSet;)V",
+			at = @At(value = "INVOKE", target = "Lnet/minecraft/block/entity/StructureBlockBlockEntity;getStructurePath()Ljava/lang/String;")
+	)
+	private static String quiltGameTest$replaceStructurePathWithName(StructureBlockBlockEntity instance) {
+		return instance.getStructureName();
+	}
+
+	@Redirect(
+			method = {"executeExport(Lnet/minecraft/server/command/ServerCommandSource;Ljava/lang/String;)I", "executeImport"},
+			at = @At(value = "NEW", target = "(Ljava/lang/String;Ljava/lang/String;)Lnet/minecraft/util/Identifier;"),
+			expect = 2
+	)
+	private static Identifier quiltGameTest$fixStructureIdentifierExportImport(String namespace, String structure) {
+		return new Identifier(structure);
+	}
+
+	@ModifyArg(
+			method = "executeImport",
+			at = @At(value = "INVOKE", target = "Ljava/nio/file/Paths;get(Ljava/lang/String;[Ljava/lang/String;)Ljava/nio/file/Path;"),
+			index = 1
+	)
+	private static String[] quiltGameTest$fixImportPath(String[] more) {
+		more[0] = more[0].replace(':', '/');
+		return more;
+	}
+
+	@ModifyArg(
+			method = "executeExport(Lnet/minecraft/server/command/ServerCommandSource;Ljava/lang/String;)I",
+			at = @At(
+					value = "INVOKE",
+					target = "Lnet/minecraft/data/dev/NbtProvider;convertNbtToSnbt(Lnet/minecraft/unmapped/C_temnquoh;Ljava/nio/file/Path;Ljava/lang/String;Ljava/nio/file/Path;)Ljava/nio/file/Path;"
+			),
+			index = 2
+	)
+	private static String quiltGameTest$fixImportPath(String structure) {
+		return structure.replace(':', '/');
+	}
+
+	@ModifyConstant(
+			method = "onCompletion",
+			constant = @Constant(stringValue = "All required tests passed :)")
+	)
+	private static String quiltGameTest$replaceSuccessMessage(String original) {
+		// You may ask why, it's simple.
+		// The original emoticon is a bit... weird.
+		// And QSL members expressed some kind of interest into replacing it.
+		// So here it is. I assure you this is a really necessary injection.
+		return "All required tests passed :3c";
+	}
```

Comment:
I love this! /gen

Reply:
oh my god you actually did it
based as all hell